spec/spec_helper.rb

```diff
@@ -50,6 +51,12 @@
 # Override setting in Spree engine: Spree::Core::MailSettings
 ActionMailer::Base.default_url_options[:host] = 'test.host'
 
+Spree::AppConfiguration.class_eval do
+  prepend Spree::TestingSupport::ConfigStub
+end
+
+Spree::Config.setup_config_stub!
```

Comment:
Just for me to understand. We don't do the regular  `allow().to receive()` due to the amount of calls `Config` receives in integration tests?

In such case, I like your idea but I'd rather implement an object with the exact same API as `Config` (duck type) and replace the original with the mock in here.

Reopening a class and then changing the behaviour of `Config` feels very risky. We could be messing up the original behaviour without even noticing.

Reply:
The reason we can’t use stubbing is that `Config` settings are read in three different ways, and it is impractical/impossible to stub all three.

The only problem with mimicing the existing config API in a mock object is that it would mean we need to come up with a way of providing all of the defaults.

I couldn’t think of a smart way of swapping out the the real config for a mock config, though I suppose I could just reassign `Spree::Config`, I didn’t even try that.

Reply:
>  I suppose I could just reassign Spree::Config, I didn’t even try that.

I believe so.